Can't run ttn-lw-stack commands after building with goreleaser

<!--
Thanks for submitting a bug report. Please fill the template below,
otherwise we will not be able to process this bug report.
-->

#### Summary
<!-- Summarize the problem in a few sentences: -->

I cannot run any `ttn-lw-stack` commands using the binaries produced by running goreleaser.

#### Steps to Reproduce
<!-- How can we reproduce the problem? -->

1. Checkout tag v3.10.7
2. Run go releaser
```
go run github.com/goreleaser/goreleaser --snapshot -f .goreleaser.snapshot.yml --rm-dist
```
3. Try a command
```
./dist/stack_linux_amd64/ttn-lw-stack start
```


<!--
Please upload relevant configuration (as .txt).
If you use the command "ttn-lw-stack config", you can redact sensitive config.
-->

#### What do you see now?
<!-- Please paste terminal output, upload logs (as .txt) or upload screenshots. -->

This is the error produced:

```
panic: reflect: call of reflect.Value.Type on zero Value

goroutine 1 [running]:
reflect.Value.Type(0x0, 0x0, 0x0, 0x25bb290, 0x2081d60)
	/usr/local/go/src/reflect/value.go:1879 +0x183
github.com/mitchellh/mapstructure.DecodeHookExec(0x216ff20, 0x25bb290, 0x0, 0x0, 0x0, 0x221a2e0, 0xc000403d88, 0x196, 0x0, 0x0, ...)
	/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.0/decode_hooks.go:46 +0x1cb
github.com/mitchellh/mapstructure.ComposeDecodeHookFunc.func1(0x2081d60, 0x40704e0, 0x98, 0x221a2e0, 0xc000403d88, 0x196, 0x0, 0xabc1757d01551b00, 0xc000ce7b90, 0xc000b4f968)
	/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.0/decode_hooks.go:67 +0xd1
github.com/mitchellh/mapstructure.DecodeHookExec(0x2147b80, 0xc00049ec60, 0x2081d60, 0x40704e0, 0x98, 0x221a2e0, 0xc000403d88, 0x196, 0x16, 0x2e, ...)
	/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.0/decode_hooks.go:50 +0x12b
github.com/mitchellh/mapstructure.(*Decoder).decode(0xc000587d98, 0xc000ce7b90, 0x2e, 0x2081d60, 0x40704e0, 0x221a2e0, 0xc000403d88, 0x196, 0x2e, 0x0)
	/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.0/mapstructure.go:416 +0x8b1
github.com/mitchellh/mapstructure.(*Decoder).decodeStructFromMap(0xc000587d98, 0xc0013dde00, 0x17, 0x2157840, 0xc000d08e70, 0x15, 0x23db400, 0xc000403d78, 0x199, 0x0, ...)
	/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.0/mapstructure.go:1357 +0x1be4
github.com/mitchellh/mapstructure.(*Decoder).decodeStruct(0xc000587d98, 0xc0013dde00, 0x17, 0x2157840, 0xc000d08e70, 0x23db400, 0xc000403d78, 0x199, 0x2157840, 0xc000d08e70)
	/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.0/mapstructure.go:1183 +0x621
github.com/mitchellh/mapstructure.(*Decoder).decode(0xc000587d98, 0xc0013dde00, 0x17, 0x2157840, 0xc000d08e70, 0x23db400, 0xc000403d78, 0x199, 0x17, 0x0)
	/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.0/mapstructure.go:439 +0x765
github.com/mitchellh/mapstructure.(*Decoder).decodeStructFromMap(0xc000587d98, 0x1f6449d, 0x2, 0x2157840, 0xc000c909f0, 0x15, 0x241a960, 0xc000403cd0, 0x199, 0x0, ...)
	/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.0/mapstructure.go:1357 +0x1be4
github.com/mitchellh/mapstructure.(*Decoder).decodeStruct(0xc000587d98, 0x1f6449d, 0x2, 0x2157840, 0xc000c909f0, 0x241a960, 0xc000403cd0, 0x199, 0x2157840, 0xc000c909f0)
	/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.0/mapstructure.go:1183 +0x621
github.com/mitchellh/mapstructure.(*Decoder).decode(0xc000587d98, 0x1f6449d, 0x2, 0x2157840, 0xc000c909f0, 0x241a960, 0xc000403cd0, 0x199, 0x94, 0x0)
	/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.0/mapstructure.go:439 +0x765
github.com/mitchellh/mapstructure.(*Decoder).decodeStructFromMap(0xc000587d98, 0x0, 0x0, 0x2157840, 0xc00099cc00, 0x15, 0x241cd20, 0xc000403000, 0x199, 0x0, ...)
	/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.0/mapstructure.go:1357 +0x1be4
github.com/mitchellh/mapstructure.(*Decoder).decodeStruct(0xc000587d98, 0x0, 0x0, 0x2157840, 0xc00099cc00, 0x241cd20, 0xc000403000, 0x199, 0x2157840, 0xc00099cc00)
	/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.0/mapstructure.go:1183 +0x621
github.com/mitchellh/mapstructure.(*Decoder).decode(0xc000587d98, 0x0, 0x0, 0x2157840, 0xc00099cc00, 0x241cd20, 0xc000403000, 0x199, 0xc0002cb300, 0xc00099cc00)
	/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.0/mapstructure.go:439 +0x765
github.com/mitchellh/mapstructure.(*Decoder).Decode(0xc000587d98, 0x2157840, 0xc00099cc00, 0x0, 0xc00049ec60)
	/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.0/mapstructure.go:374 +0xf0
go.thethings.network/lorawan-stack/v3/pkg/config.(*Manager).Unmarshal(0xc0008f2c00, 0x2188d80, 0xc000403000, 0xc0006cbc00, 0xc000dcfab0)
	/tmp/build/80754af9/lorawan-stack/pkg/config/config.go:235 +0x2bf
go.thethings.network/lorawan-stack/v3/cmd/ttn-lw-stack/commands.glob..func7(0x3fe9e00, 0x406e568, 0x0, 0x0, 0x0, 0x0)
	/tmp/build/80754af9/lorawan-stack/cmd/ttn-lw-stack/commands/root.go:60 +0x7d
github.com/spf13/cobra.(*Command).execute(0x3fe9e00, 0x406e568, 0x0, 0x0, 0x3fe9e00, 0x406e568)
	/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:829 +0x55e
github.com/spf13/cobra.(*Command).ExecuteC(0x3fe9b60, 0x0, 0x2068420, 0xc0000540b8)
	/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:958 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:895
main.main()
	/tmp/build/80754af9/lorawan-stack/cmd/ttn-lw-stack/main.go:26 +0x2d
```

#### What do you want to see instead?
<!-- Please add some examples or mock-ups if applicable. -->

The command doing something. Not a reflection error.

#### Environment
<!--
Your environment: OS/Browser/Gateway/Device/...? Versions? IDs/EUIs?
Paste the output of "ttn-lw-cli version" or "ttn-lw-stack version" if applicable.
-->
```
# ./dist/stack_linux_amd64/ttn-lw-stack version
panic: reflect: call of reflect.Value.Type on zero Value

goroutine 1 [running]:
reflect.Value.Type(0x0, 0x0, 0x0, 0x25bb290, 0x2081d60)
	/usr/local/go/src/reflect/value.go:1879 +0x183
github.com/mitchellh/mapstructure.DecodeHookExec(0x216ff20, 0x25bb290, 0x0, 0x0, 0x0, 0x221a2e0, 0xc00066c588, 0x196, 0x0, 0x0, ...)
	/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.0/decode_hooks.go:46 +0x1cb
github.com/mitchellh/mapstructure.ComposeDecodeHookFunc.func1(0x2081d60, 0x40704e0, 0x98, 0x221a2e0, 0xc00066c588, 0x196, 0x0, 0x7712959e361e3e00, 0xc000a7de00, 0xc0009f3968)
	/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.0/decode_hooks.go:67 +0xd1
github.com/mitchellh/mapstructure.DecodeHookExec(0x2147b80, 0xc000ab33a0, 0x2081d60, 0x40704e0, 0x98, 0x221a2e0, 0xc00066c588, 0x196, 0x16, 0x2e, ...)
	/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.0/decode_hooks.go:50 +0x12b
github.com/mitchellh/mapstructure.(*Decoder).decode(0xc000a72ce8, 0xc000a7de00, 0x2e, 0x2081d60, 0x40704e0, 0x221a2e0, 0xc00066c588, 0x196, 0x2e, 0x0)
	/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.0/mapstructure.go:416 +0x8b1
github.com/mitchellh/mapstructure.(*Decoder).decodeStructFromMap(0xc000a72ce8, 0xc000c06e40, 0x17, 0x2157840, 0xc0007204e0, 0x15, 0x23db400, 0xc00066c578, 0x199, 0x0, ...)
	/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.0/mapstructure.go:1357 +0x1be4
github.com/mitchellh/mapstructure.(*Decoder).decodeStruct(0xc000a72ce8, 0xc000c06e40, 0x17, 0x2157840, 0xc0007204e0, 0x23db400, 0xc00066c578, 0x199, 0x2157840, 0xc0007204e0)
	/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.0/mapstructure.go:1183 +0x621
github.com/mitchellh/mapstructure.(*Decoder).decode(0xc000a72ce8, 0xc000c06e40, 0x17, 0x2157840, 0xc0007204e0, 0x23db400, 0xc00066c578, 0x199, 0x17, 0x0)
	/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.0/mapstructure.go:439 +0x765
github.com/mitchellh/mapstructure.(*Decoder).decodeStructFromMap(0xc000a72ce8, 0x1f6449d, 0x2, 0x2157840, 0xc0006c2b10, 0x15, 0x241a960, 0xc00066c4d0, 0x199, 0x0, ...)
	/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.0/mapstructure.go:1357 +0x1be4
github.com/mitchellh/mapstructure.(*Decoder).decodeStruct(0xc000a72ce8, 0x1f6449d, 0x2, 0x2157840, 0xc0006c2b10, 0x241a960, 0xc00066c4d0, 0x199, 0x2157840, 0xc0006c2b10)
	/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.0/mapstructure.go:1183 +0x621
github.com/mitchellh/mapstructure.(*Decoder).decode(0xc000a72ce8, 0x1f6449d, 0x2, 0x2157840, 0xc0006c2b10, 0x241a960, 0xc00066c4d0, 0x199, 0x94, 0x0)
	/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.0/mapstructure.go:439 +0x765
github.com/mitchellh/mapstructure.(*Decoder).decodeStructFromMap(0xc000a72ce8, 0x0, 0x0, 0x2157840, 0xc000980fc0, 0x15, 0x241cd20, 0xc00066b800, 0x199, 0x0, ...)
	/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.0/mapstructure.go:1357 +0x1be4
github.com/mitchellh/mapstructure.(*Decoder).decodeStruct(0xc000a72ce8, 0x0, 0x0, 0x2157840, 0xc000980fc0, 0x241cd20, 0xc00066b800, 0x199, 0x2157840, 0xc000980fc0)
	/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.0/mapstructure.go:1183 +0x621
github.com/mitchellh/mapstructure.(*Decoder).decode(0xc000a72ce8, 0x0, 0x0, 0x2157840, 0xc000980fc0, 0x241cd20, 0xc00066b800, 0x199, 0xc0005cd980, 0xc000980fc0)
	/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.0/mapstructure.go:439 +0x765
github.com/mitchellh/mapstructure.(*Decoder).Decode(0xc000a72ce8, 0x2157840, 0xc000980fc0, 0x0, 0xc000ab33a0)
	/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.0/mapstructure.go:374 +0xf0
go.thethings.network/lorawan-stack/v3/pkg/config.(*Manager).Unmarshal(0xc000633800, 0x2188d80, 0xc00066b800, 0x7f5752d503f0, 0xc000dd9ab0)
	/tmp/build/80754af9/lorawan-stack/pkg/config/config.go:235 +0x2bf
go.thethings.network/lorawan-stack/v3/cmd/ttn-lw-stack/commands.glob..func7(0xc000210dc0, 0x406e568, 0x0, 0x0, 0x0, 0x0)
	/tmp/build/80754af9/lorawan-stack/cmd/ttn-lw-stack/commands/root.go:60 +0x7d
github.com/spf13/cobra.(*Command).execute(0xc000210dc0, 0x406e568, 0x0, 0x0, 0xc000210dc0, 0x406e568)
	/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:829 +0x55e
github.com/spf13/cobra.(*Command).ExecuteC(0x3fe9b60, 0x0, 0x2068420, 0xc0000540b8)
	/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:958 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:895
main.main()
	/tmp/build/80754af9/lorawan-stack/cmd/ttn-lw-stack/main.go:26 +0x2d
```
```
# ./dist/cli_linux_amd64/ttn-lw-cli version
The Things Network Command-line Interface: ttn-lw-cli
Version:             v3.10.7-SNAPSHOT-ecf52d6b5
Build date:          2021-02-03T01:44:28Z
Git commit:          ecf52d6b5
Go version:          go1.15.7
OS/Arch:             linux/amd64
```

#### How do you propose to implement this?
<!-- Please think about how this could be fixed. -->

N/A

#### How do you propose to test this?
<!-- Please think about how this is verified as fixed. -->

N/A

#### Can you do this yourself and submit a Pull Request?
<!-- You can also @mention experts if you need help with this. -->

No.

--- a/DEVELOPMENT.md
+++ b/DEVELOPMENT.md
@@ -1043,7 +1043,25 @@ It is also possible to use `go build`, or release snapshots, as described below.
 
 ## Releasing
 
-You can build a release snapshot with `cd tools && go run github.com/goreleaser/goreleaser --snapshot`.
+The Things Stack uses [GoReleaser](https://goreleaser.com/) for releases. If you want to build a release (snapshot), you first need to build GoReleaser:
+
+```bash
+$ pushd tools
+$ go build -o bin/goreleaser github.com/goreleaser/goreleaser
+$ popd
+```
+
+The command for building a release snapshot is:
+
+```bash
+$ tools/bin/goreleaser --snapshot -f .goreleaser.snapshot.yml --rm-dist
+```
+
+The command for building a full release is:
+
+```bash
+$ tools/bin/goreleaser -f .goreleaser.release.yml --rm-dist
+```
 
 > Note: You will at least need to have [`rpm`](http://rpm5.org/) and [`snapcraft`](https://snapcraft.io/) in your `PATH`.
 

